extensions/amp-connatix-player/0.1/amp-connatix-player.js

```diff
@@ -71,10 +105,85 @@ export class AmpConnatixPlayer extends AMP.BaseElement {
         // Ignore messages from other iframes.
         return;
       }
-      // Player wants to close because the user interracted on its close button
-      if (getData(e) === 'cnx_close') {
-        this.destroyPlayerFrame_();
-        this.attemptCollapse();
+      const dataString = getData(e);
+      const dataJSON = tryParseJson(dataString) || dict();
+
+      if (dataJSON['event'] !== 'command') {
+        return;
+      }
+
+      switch (dataJSON['func']) {
+        // Player wants to close because the user interacted on its close button
+        case 'cnxClose': {
+          this.destroyPlayerFrame_();
+          this.attemptCollapse();
+          break;
+        }
+        // Player rendered
+        case 'cnxPlayerRendered': {
+          this.playerReadyResolver_(this.iframe_);
+          break;
+        }
+      }
+    });
+  }
+
+  /**
+   * Binds to amp-consent
+   * @private
+   */
+  bindToAmpConsent_() {
+    const consentPolicyId = super.getConsentPolicy() || 'default';
+    const consentPolicyStatePromise = getConsentPolicyState(
+      this.element,
+      consentPolicyId
+    );
+    const consentPolicyInfoPromise = getConsentPolicyInfo(
+      this.element,
+      consentPolicyId
+    );
+    const consentPolicySharedDataPromise = getConsentPolicySharedData(
+      this.element,
+      consentPolicyId
+    );
+    const consentMetadataPromise = getConsentMetadata(
+      this.element,
+      consentPolicyId
+    );
+
+    Promise.allSettled([
```

Comment:
I'm not certain we should use `Promise.allSettled` due to incompatibility with IE11.

@jridgewell This appears to be the first usage in client extensions. What's your take?

Reply:
Via @jridgewell on Slack: We do need a polyfill or helper method for this usage.

Here's an example implementation as a exportable helper (good for this PR).

```javascript
/**
 * @param {!Array<T>} promises
 * @return {!Promise<!Array<{
 *   status: string,
 *   value: (T|undefined),
 *   reason: *,
 * }>>}
 * @template T
 */
export function allSettled(promises) {
  /**
   * @param {*} value
   * @return {{status: string, value: *}}
   */
  function onFulfilled(value) {
    return {status: 'fulfilled', value};
  }
  /**
   * @param {*} reason
   * @return {{status: string, reason: *}}
   */
  function onRejected(reason) {
    return {status: 'rejected', reason};
  }
  return Promise.all(
    promises.map((promise) => {
      return promise.then(onFulfilled, onRejected);
    })
  );
}
```

Reply:
Thank you for your help guys